Sources/SwiftKueryORM/Database.swift

```diff
@@ -45,12 +48,29 @@ public class Database {
     /// connection generator
     private enum ConnectionStrategy {
         case pool(ConnectionPool)
-        case generator(() -> Connection?)
+        case generator(((@escaping DatabaseTask)) -> ())
     }
 
     private let connectionStrategy: ConnectionStrategy
 
-    /// Constructor for a single connection which becomes a connection pool
+    /**
+     Create a Database instance which uses a single connection to perform each operation.
+     The connection is wrapped in a connection pool which prevents any concurrent operations being carried out on the connection
```

Comment:
I don't think we need to describe this internal implementation detail.  From the user's perspective, they are just creating a connection to a database.  We should also mention the trade-offs of doing this vs. the pool / generator options.  Perhaps we could reword this block as follows:
> Create a Database instance which uses a single connection to perform each operation.  The connection will remain open for the lifetime of the Database.
> It is safe for you to issue concurrent operations on the database: the ORM will ensure those operations are executed sequentially.
> This connection strategy provides lower latency requests, as a new connection does not need to be established for each operation, and has a small resource overhead.  However, scalability is limited as all database operations are serialized.